Libraries/LibWeb/HTML/HTMLButtonElement.cpp

```diff
@@ -46,6 +47,13 @@ WebIDL::ExceptionOr<void> HTMLButtonElement::set_type(String const& type)
     return set_attribute(HTML::AttributeNames::type, type);
 }
 
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validity
+GC::Ref<ValidityState const> HTMLButtonElement::validity() const
+{
+    auto& realm = this->realm();
+    return realm.create<ValidityState>(realm, static_cast<FormAssociatedElement const*>(this));
+}
+
```

Comment:
This method should only need to be implemented once in `FormAssociatedElement`.

Reply:
I tried doing that, but the IDL generator didn't like it.
Is there some way to do this already or will I need to modify the IDL generator?

Reply:
This change shouldn't affect the IDL generator. Does doing this work for you https://github.com/LadybirdBrowser/ladybird/commit/f74d35250d695013b9fe8229e4d0246777ac006d?

Reply:
That... does work.
I must've messed something else up then when trying it myself before, sorry.